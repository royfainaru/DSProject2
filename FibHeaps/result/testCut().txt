╮
├── ╭────────────────╮
│   │ Key: 9         │
│   ├────────────────┤
│   │ Rank: 0        │
│   │ Marked: false  │
│   │ Parent: (null) │
│   │ Next: 2        │
│   │ Prev: (null)   │
│   │ Child: (null)  │
│   ╰────────────────╯
╰── ╭────────────────╮
    │ Key: 2         │
    ├────────────────┤
    │ Rank: 3        │
    │ Marked: false  │
    │ Parent: (null) │
    │ Next: (null)   │
    │ Prev: 9        │
    │ Child: 4       │
    ╰────────────────╯
    ├── ╭───────────────╮
    │   │ Key: 4        │
    │   ├───────────────┤
    │   │ Rank: 2       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 10      │
    │   │ Prev: (null)  │
    │   │ Child: 6      │
    │   ╰───────────────╯
    │   ├── ╭───────────────╮
    │   │   │ Key: 6        │
    │   │   ├───────────────┤
    │   │   │ Rank: 1       │
    │   │   │ Marked: false │
    │   │   │ Parent: 4     │
    │   │   │ Next: 7       │
    │   │   │ Prev: (null)  │
    │   │   │ Child: 8      │
    │   │   ╰───────────────╯
    │   │   ╰── ╭───────────────╮
    │   │       │ Key: 8        │
    │   │       ├───────────────┤
    │   │       │ Rank: 0       │
    │   │       │ Marked: false │
    │   │       │ Parent: 6     │
    │   │       │ Next: (null)  │
    │   │       │ Prev: (null)  │
    │   │       │ Child: (null) │
    │   │       ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 7        │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 4     │
    │       │ Next: (null)  │
    │       │ Prev: 6       │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ├── ╭───────────────╮
    │   │ Key: 10       │
    │   ├───────────────┤
    │   │ Rank: 1       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 3       │
    │   │ Prev: 4       │
    │   │ Child: 11     │
    │   ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 11       │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 10    │
    │       │ Next: (null)  │
    │       │ Prev: (null)  │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ╰── ╭───────────────╮
        │ Key: 3        │
        ├───────────────┤
        │ Rank: 0       │
        │ Marked: false │
        │ Parent: 2     │
        │ Next: (null)  │
        │ Prev: 10      │
        │ Child: (null) │
        ╰───────────────╯
