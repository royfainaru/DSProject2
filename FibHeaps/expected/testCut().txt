╮
├── ╭────────────────╮
│   │ Key: 9         │
│   ├────────────────┤
│   │ Rank: 0        │
│   │ Marked: false  │
│   │ Parent: (null) │
│   │ Next: 2        │
│   │ Prev: 2        │
│   │ Child: (null)  │
│   ╰────────────────╯
╰── ╭────────────────╮
    │ Key: 2         │
    ├────────────────┤
    │ Rank: 3        │
    │ Marked: false  │
    │ Parent: (null) │
    │ Next: 9        │
    │ Prev: 9        │
    │ Child: 4       │
    ╰────────────────╯
    ├── ╭───────────────╮
    │   │ Key: 4        │
    │   ├───────────────┤
    │   │ Rank: 2       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 10      │
    │   │ Prev: 3       │
    │   │ Child: 6      │
    │   ╰───────────────╯
    │   ├── ╭───────────────╮
    │   │   │ Key: 6        │
    │   │   ├───────────────┤
    │   │   │ Rank: 1       │
    │   │   │ Marked: false │
    │   │   │ Parent: 4     │
    │   │   │ Next: 7       │
    │   │   │ Prev: 7       │
    │   │   │ Child: 8      │
    │   │   ╰───────────────╯
    │   │   ╰── ╭───────────────╮
    │   │       │ Key: 8        │
    │   │       ├───────────────┤
    │   │       │ Rank: 0       │
    │   │       │ Marked: false │
    │   │       │ Parent: 6     │
    │   │       │ Next: 8       │
    │   │       │ Prev: 8       │
    │   │       │ Child: (null) │
    │   │       ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 7        │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 4     │
    │       │ Next: 6       │
    │       │ Prev: 6       │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ├── ╭───────────────╮
    │   │ Key: 10       │
    │   ├───────────────┤
    │   │ Rank: 1       │
    │   │ Marked: false │
    │   │ Parent: 2     │
    │   │ Next: 3       │
    │   │ Prev: 4       │
    │   │ Child: 11     │
    │   ╰───────────────╯
    │   ╰── ╭───────────────╮
    │       │ Key: 11       │
    │       ├───────────────┤
    │       │ Rank: 0       │
    │       │ Marked: false │
    │       │ Parent: 10    │
    │       │ Next: 11      │
    │       │ Prev: 11      │
    │       │ Child: (null) │
    │       ╰───────────────╯
    ╰── ╭───────────────╮
        │ Key: 3        │
        ├───────────────┤
        │ Rank: 0       │
        │ Marked: false │
        │ Parent: 2     │
        │ Next: 4       │
        │ Prev: 10      │
        │ Child: (null) │
        ╰───────────────╯
